### DISCUSSION TITLE
TGS request result: Server not found in Kerberos database

### DISCUSSION THREAD

**Discussion by shrimallepudi** [edenhill/librdkafka#3896] [Q&A]
I'm using librdkafka 1.8.2 with confluent-kafka 1.8.2. I've configured my kafka producer as below:

   'bootstrap.servers': <servers>,
   'client.id': socket.gethostname(),
    'debug': 'security,broker',
    'security.protocol': 'SASL_PLAINTEXT',
    'sasl.kerberos.service.name': <service_name>,
    'sasl.kerberos.principal': <principal_id>,
    'sasl.mechanism': 'GSSAPI', 
    'sasl.kerberos.keytab': <kt_file>,
    'sasl.kerberos.kinit.cmd': 'kinit -R -t "%{sasl.kerberos.keytab}" -k %{sasl.kerberos.principal} || kinit -t "%{sasl.kerberos.keytab}" -k %{sasl.kerberos.principal}',
    'enable.auto.commit': True

When I run my python producer script, it throws error saying '**_TGS request result: -XXXXX/Server not found in Kerberos database_**'. Can someone please let me know what's wrong? I'm using Python 3.6

PS: I'm able to publish messages to the clusters/brokers via console-producer.sh.

**Comment by rolandyoung** [comment]
There is something wrong with your Kerberos configuration options. Check that kinit works with the parameters shown.

**Reply by shrimallepudi** [reply]
I tried that and it worked absolutely fine. I ran klist too, right after and it showed me the relevant details

**Reply by rolandyoung** [reply]
That error message definitely means that the given ticket cannot be obtained with the given kinit parameters.
So either there is a bug in libsasl, or there is something in your Kerberos environment that you have not noticed. This is not an issue that can easily be debugged in this forum, especially given that you cannot (for good reasons) show us the exact text of either the configuration options or the error messages.

**Reply by shrimallepudi** [reply]
I understand. I'm sure that we have all the necessary gssapi/sasl packages installed, but still, a validation would always be helpful. Could you please list down the libraries/packages I need to get installed on our servers?

**Reply by rolandyoung** [reply]
When you raised this issue, you deleted the checklist that says "**IMPORTANT**: We will close issues where the checklist has not been completed." Please fill that out.
Having said that, you would probably get a different error if you had a library missing. That error indicates that the correct library is being invoked, but it is unable to obtain the required ticket.

**Reply by shrimallepudi** [reply]
This was actually raised as a discussion thread rather than an issue, which is why I didn't see any option to add checklist. Where do I need to do that?

**Reply by rolandyoung** [reply]
Sorry, my mistake. To see the checklist, you could start to raise an issue but close the browser tab before actually doing so. I'm sorry, but I am going to have to leave this thread alone now. I'm not a librdkafka maintainer, but I recognised a Kerberos error and tried to help by checking that you had done the kinit / klist checks.

**Reply by shrimallepudi** [reply]
Thank you for stepping in, Roland. I appreciate it :)
